lib/std/x/net/tcp.zig

```diff
@@ -89,41 +112,46 @@ pub const Client = struct {
         return self.socket.connect(address.into());
     }
 
-    /// Read data from the socket into the buffer provided. It returns the
-    /// number of bytes read into the buffer provided.
-    pub fn read(self: Client, buf: []u8) !usize {
-        return self.socket.read(buf);
+    /// Extracts the error set of a function.
+    /// TODO: remove after Socket.{read, write} error unions are well-defined across different platforms
+    fn ErrorSetOf(comptime Function: anytype) type {
```

Comment:
this looks like something that belongs in `std.meta`

Reply:
It's just a temporary helper method for now, though would you propose that we have a method like this in `std.meta` in general?